boa/network.py

```diff
@@ -132,16 +132,16 @@ def _rpc_has_snapshot(self):
     def anchor(self):
         if not self._rpc_has_snapshot:
             raise RuntimeError("RPC does not have `evm_snapshot` capability!")
+        block_number = self.evm.patch.block_number
+        snapshot_id = self._rpc.fetch("evm_snapshot", [])
         try:
-            block_id = self.evm.patch.block_id
```

Comment:
i prefer block_id since theoretically it could be a number or hash. is there some reason to change it here?

Reply:
It was changed in this PR: https://github.com/vyperlang/titanoboa/pull/134/files#diff-7fc173b3ee71c685d73370cbf30e285056ce409ae238c1c5d2d7ce9c97bcd04bR136

Reply:
let's separate out this fix into another PR